runtime/lua/vim/_editor.lua

```diff
@@ -568,14 +568,20 @@ end
 ---
 ---@param msg string Content of the notification to show to the user.
 ---@param level integer|nil One of the values from |vim.log.levels|.
----@param opts table|nil Optional parameters. Unused by default.
+---@param opts table|nil Options:
+---   - history: (boolean) if true, add to |message-history|.
 function vim.notify(msg, level, opts) -- luacheck: no unused args
+  opts = opts or {}
```

Comment:
maybe there can just `opts = opts or { history = true} ` instead of create `local history `. then use `opts.history`  and add a  `vim.validate` for `history` then you don't need do `type` assertion.

Reply:
hi, @glepnir thanks for reviewing, but I do not know how to use `vim.validate`, I just read the `:h vim.validate`, it only shows how to check the type of argv, for example:

```
function vim.notify(msg, level, opts) -- luacheck: no unused args
   vim.validate(
         opts = {opts, 't'}
   )
```

but I do not know how to check the type of `opts.history` in vim.validate.